internal/codegen/cel.go

```diff
@@ -79,26 +81,35 @@ type CELCondition struct {
 	ast *cel.Ast
 }
 
-func (cc *CELCondition) Program() (cel.Program, error) {
-	return cc.env.Program(cc.ast)
+func (cc *CELCondition) Program(vars ...*exprpb.Decl) (cel.Program, error) {
+	if len(vars) == 0 {
+		return cc.env.Program(cc.ast)
+	}
+	opts := append([]cel.EnvOption{cel.Declarations(vars...)}, NewCELEnvOptions()...)
+	env, err := cel.NewEnv(opts...)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Is it possible to use [`env.Extend`](https://pkg.go.dev/github.com/google/cel-go@v0.8.0/cel#Env.Extend) here?

Reply:
Yes, thanks for pointing it out.